source/fe/fe_system.cc

```diff
@@ -1161,10 +1074,11 @@ compute_fill_one_base (const Mapping<dim,spacedim>                      &mapping
                   output_data.shape_gradients[out_index+s][q] =
                     base_data.shape_gradients[in_index+s][q];
 
-              // _we_ handle computation of second derivatives, so the
-              // base elements should not have computed them!
-              Assert (!(base_flags & update_hessians),
-                      ExcInternalError());
+              if (base_flags & update_hessians)
+                for (unsigned int q=0; q<n_q_points; ++q)
+                  output_data.shape_hessians[out_index+s][q] =
+                    base_data.shape_hessians[in_index+s][q];
+
```

Comment:
@bangerth I'm fine with the patch as soon as the line change is fixed, so I let you merge if you don't have further comments. I assume this patch will conflict with #1318 so I'd rather merge this one first (I'd say @maieneuro has resolved enough merge conflicts for this patch) and resolve possible conflicts in #1318 rather than here.